### PR TITLE
Auth implementation

### PR DESCRIPTION
**Pull request by fahrinh**
- For web app, cookie is recommended place to store sensitive data like token.
But, to send cookie without setup CORS on backend, web app (frontend and backend) must be served under same domain. For that reason, this PR includes `setupProxy.js` on frontend to forward all requests with prefix `localhost:3000/api/**` to backend endpoint `locahost:8089/api/**`.
So we can think `localhost:3000` is main domain (with port) to serve web app and endpoint.
`REACT_APP_API_URL` in `.env` frontend become unnecessary because of this setup.
- To allow serving requests from public (like mobile app), CORS config on backend is set `AllowOrigins: *` 
- Endpoints with prefix `/api/**` are protected with JWT middleware
- These endpoints are public (not protected):
  - `/auth/google/login`
  - `/auth/google/callback`
- If Google auth is successful, backend will redirect and set these two cookies (that stored JWT token):
  - `secure_token`: This is `httpOnly` cookie so only backend can read. JWT middleware (backend) will read this cookie to verify JWT token
  - `token`: HotstoneUI (frontend) will read this cookie
  - This trick come from: https://stackoverflow.com/a/51965652

Related issue: #5

### CODE REVIEW

File: ui-v2/src/App.js [hunk 1]
```diff
@@ -1,16 +1,14 @@
-import React from 'react';
-import { BrowserRouter, Route, Switch } from 'react-router-dom';
-import { Spin } from 'antd';
-import { AuthProvider, PrivateRoute } from 'components';
-import './App.css';
+import React from "react";
```
Comment (hawarir) [inline ui-v2/src/App.js:1]: Can you revert all single-quote to double-quote fix?

File: ui-v2/src/setupProxy.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const { createProxyMiddleware } = require("http-proxy-middleware");
+
+module.exports = function(app) {
```
Comment (hawarir) [inline ui-v2/src/setupProxy.js:3]: Where are we using this function and why do we have a hard-coded URL in it?

File: ui-v2/src/App.js (hunk 1 above)
Comment (fahrinh) [reply ui-v2/src/App.js:1]: For all `*.js` ?

File: ui-v2/src/setupProxy.js (hunk 2 above)
Comment (fahrinh) [reply ui-v2/src/setupProxy.js:3]: Forgot to mention the doc: https://create-react-app.dev/docs/proxying-api-requests-in-development/#configuring-the-proxy-manually

> You do not need to import this file anywhere. It is automatically registered when you start the development server.

That's only for development. It is not included in the production build.

File: ui-v2/src/App.js (hunk 1 above)
Comment (hawarir) [reply ui-v2/src/App.js:1]: Well, just the one that you unintentionally change should be fine. I believe you can just use the ESLint configuration to fix the quote.

File: ui-v2/src/setupProxy.js (hunk 2 above)
Comment (hawarir) [reply ui-v2/src/setupProxy.js:3]: Ah I see, it should be fine then

File: ui-v2/src/App.js (hunk 1 above)
Comment (fahrinh) [reply ui-v2/src/App.js:1]: Done on the latest commit

**Review by hawarir — APPROVED** [review]

**Pull request merged**